config/sidekiq.yml

```diff
@@ -0,0 +1,7 @@
+schedule:
+  periodic_task_reminder:
+    class: PeriodicTaskReminder
+    every: 1h
+  task_notification:
+    class: TaskNotification
+    every: 5m
```

Comment:
ultimately we would want a much shorter interval, e.g. 10s-30s

also not entirely sure how you planned this, but the timer should start and re-start for every measure change or measure-relationship change, and eventually send once expired. simply sending everything at set intervals would not be ideal

Reply:
Yep, that's right, it asks the `TaskNotification` job to find all tasks that match the criteria for sending (i.e. ones that happened at least 5 minutes ago and haven't been notified yet - based on an entry in the `task_notifications` table).

Reply:
hmm still not sure I fully understand...

* does this mean the task runs constantly and constantly checks for measures requiring notifications or is started only once a measure is updated? (it should be the latter)
* why is it called "every"?

Reply:
so I thought the idea was: every time an activity is updated (or one of its connections) a noitification for that measure is triggered, replacing any previously scheduled notifications for that measure, and waiting 20s or so until it is eventually sent (given the chance for later changes to replace it and reset the timer)

Reply:
>    does this mean the task runs constantly and constantly checks for measures requiring notifications or is started only once a measure is updated? (it should be the latter)

Yes, it means the former. We could do it the other way that you describe where every time something happens we schedule a notification that cancels itself but if something goes wrong then we will end up with notifications not being sent and no way for us to re-queue those because we have no way of knowing which ones need to be notified.

>    why is it called "every"?

"every" here means how frequently the scheduler triggers the check, i.e. in this one it checks every 5 minutes whether there are any notifications to send, but this can be as low as every 1 minute.

Thanks for all the feedback on this, it's still early in the implementation so the right time to be asking these questions.